### DISCUSSION TITLE
Identify, incorporate a slugify routine into utils

### DISCUSSION THREAD

**Discussion by stucka** [biglocalnews/clean-scraper#79] [Ideas]
I don't know what's necessary as far as licensing goes. Django's has worked for me in the past, but then that part of it goes under a different license.

**Comment by zstumgoren** [comment]
Let's just add our own simple slugify function in a utils.py. That way we can avoid the heavy dependency on Django and sidestep licensing issues (we typically use Apache 2.0)

**Reply by newsroomdev** [reply]
@stucka was there a particular aspect or feature of Django slugify which we could incorporate ourselves quickly? Here's `slugify's` [source code](https://github.com/django/django/blob/b99c608ea10cabc97a6b251cdb6e81ef2a83bdcf/django/utils/text.py#L452-L469) (it's rather short). We might be able to lightly refactor it and add it as `utils/slugify.py` without importing Django. (I'd recommend an inline comment acknowledging, thanking, and linking to the source code.)

```py
def slugify(value, allow_unicode=False):
    """
    Convert to ASCII if 'allow_unicode' is False. Convert spaces or repeated
    dashes to single dashes. Remove characters that aren't alphanumerics,
    underscores, or hyphens. Convert to lowercase. Also strip leading and
    trailing whitespace, dashes, and underscores.
    """
    value = str(value)
    if allow_unicode:
        value = unicodedata.normalize("NFKC", value)
    else:
        value = (
            unicodedata.normalize("NFKD", value)
            .encode("ascii", "ignore")
            .decode("ascii")
        )
    value = re.sub(r"[^\w\s-]", "", value.lower())
    return re.sub(r"[-\s]+", "-", value).strip("-_")
```

**Reply by stucka** [reply]
I have no deep knowledge here -- just thinking we need a slugify routine, and it should be standardized. It can be an external dependency as far as I'm concerned, like python-slugify, but I don't know what effect if any that has on licensing.

I also see a very simple answer here:
https://stackoverflow.com/a/8366771
